_posts/2020-05-20-hive-acid.md

```diff
@@ -0,0 +1,320 @@
+| Flat                         | TPCDS store_sales scale 3000 table, 8.6B rows                                                                                          | 0                 | 0                        |
+| Only Base                    | Hive ACID store_sales scale 3000 table: 8.6B rows                                                                                      | 0                 | 0                        |
+| Base + 1-Delete              | Derived from “Only Base” with rows having customer_id=100 deleted by 1 DELETE query: 347 deleted entries                               | 0                 | 1                        |
+| Base + 1-Delete + 1-Insert   | Derived from “Base + 1 Delete” with deleted rows added back by 1 INSERT query: 347 deleted entries + 347 inserted entries              | 1                 | 1                        |
+| Base + 5-Deletes             | Derived from “Only Base” with rows for 5 customer_ids deleted by 5 DELETE queries: 1355 rows deleted                                   | 0                 | 5                        |
+| Base + 5-Deletes + 5-Inserts | Derived from “Base + 1 Delete” with deleted rows added back by 5 INSERT queries: 1355 deleted entries + 1355 inserted entries          | 5                 | 5                        |
+
+Following is the result of these tests, ran on a cluster with 5 c3.4xlarge
+machines on AWS:
+![](/assets/blog/hive-acid/perf.png)
+
+It was seen that there is an impact of deleted rows on read performance, which
+is expected as the work for the reader increases in this case. But with
+predicates in place, this impact was reduced as the amount of data to be read
+goes down.
+
+# Ongoing and Future work
+There has been ongoing work on the Hive ACID integration and some improvements
+are planned in future, notably:
+* Bucketed Hive ACID table support has been added ([#1591](https://github.com/prestosql/presto/pull/1591))
+* Support for original files is in progress ([#2930](https://github.com/prestosql/presto/pull/2930)),
+this will allow Presto to read tables that were converted to Hive Transactional
+table at some point after having non-transactional data
+* Write support will be taken up in future ([#1956](https://github.com/prestosql/presto/issues/1956))
+* There is ongoing work on Hive side for ACID on Parquet format. Once that
+lands, Presto’s implementation will be extended to support Parquet too.
+
+# Acknowledgements and Conclusion
+Thanks to the folks who helped out in the development of this feature:
+[Abhishek Somani](https://www.linkedin.com/in/abhishek-somani-a946aa1b) provided
+continuous guidance on internals of Hive ACID,
+[Dain](https://www.linkedin.com/in/dainsundstrom) helped out with simplifying
+ORC reader, code refinement and helped in multiple rounds of reviews along with
+[Piotr](https://www.linkedin.com/in/piotr-findeisen-070846b).
```

Comment:
BTW if there is nothing special you want to mention regarding myself, you can just skip me.

Reply:
rewording this. It didn't came out as intended, wanted to acknowledge both you and Dain for refinements and reviews.